src/WebJobs.Extensions.DurableTask/LinuxAppServiceLogger.cs

```diff
@@ -129,9 +227,60 @@ private string GenerateJsonStr(EventWrittenEventArgs eventData)
                 json.Add("RelatedActivityId", eventData.RelatedActivityId);
             }
 
-            // Generate string-representation of JSON. Also remove newlines.
-            string jsonString = json.ToString(Newtonsoft.Json.Formatting.None);
-            jsonString = jsonString.Replace("\n", "\\n").Replace("\r", "\\r");
+            string jsonString = "";
+            if (!this.writeToConsole)
+            {
+                // This path supports the legacy regex-based parser in Linux Dedicated.
+                // In the future, we'll be able to remove this and use JSON logging only
+                IList<string> jsonKeys = json.Properties().Select(p => p.Name).ToList();
+                List<string> regexValues = new List<string>();
+
+                foreach (string column in this.orderedRegexCol)
+                {
+                    string val = "";
+  
+                    if (json.TryGetValue(column, out JToken val))
+                    {
+                        try
+                        {
+                            // We escape a few special characters to avoid parsing problems:
+                            // (1) Escaping newline-like characters such as \n and \r to keep logs being 1 line
+                            // (2) Escaping double-quotes (") to be single-quotes (') because some fields in our regex string are
+                            //     deliniated by double-quotes. Note, this was a convention copied from the Functions Host regex
+                            //     which also uses double-quotes to capture columns that may contain commas inside them.
+                            // (3) Escaping commas (",") for ";;" because commas separate our columns and so they have the potential to
+                            //     disrupt parsing.
+                            // Note: In retrospective, perhaps the regex-string could have been designed to avoid these awkward
+                            // parsing problems, but it wasn't because (1) it followed conventions from other regex-strings in our system
+                            // and because we asssumed we'd have a JSON-based logger that would have avoided these problems.
+                            val = (string)val;
```

Comment:
ah, these two variables have the same name. That's not great. I'll give them different ones. Also, I suppose JTokens can always be casted to string, so this try-catch and probably be removed